transport/grpc/outbound.go

```diff
@@ -127,7 +128,7 @@ func (o *Outbound) Call(ctx context.Context, request *transport.Request) (*trans
 		return nil, err
 	}
 	return &transport.Response{
-		Body:                 ioutil.NopCloser(bytes.NewBuffer(responseBody)),
+		Body:                 io.NopCloser(bytes.NewReader(responseBody)),
```

Comment:
iirc bytes buffer does not copy the buffer internally and this is used for a reader with `nopcloser`. If this isn't required, can we avoid making any changes in gRPC?

Reply:
IMO this change tightens the contracts between layers. By returning a `bytes.NewBuffer` in the response Body, we are allowing a caller to use type conversion and modify the underlying data. Is there any risk in using a bytes.Reader instead of bytes.Buffer?

Reply:
I'll move this to a separate PR.

Reply:
https://github.com/yarpc/yarpc-go/pull/2233